### PR TITLE
feat: [M3-6525] - Storybook story POC

### PR DESCRIPTION
**Pull request by abailly-akamai**
## Description 📝
This PR puts forward a POC to build stories with storybook 7 & autodocs.
It takes advantage of new features and offers improvements to be made to our components to increase both their self documentation and compatibility with S7.

Please go through the self review to get to the implementation details.

## Preview 📷
Screenshot 2023-04-27 at 2 24 10 PM

Screenshot 2023-04-27 at 2 30 26 PM

## How to test 🧪
1. get branch and run `yarn install` & `rm -rf node_modules/.vite`
2. run `yarn storybook`
3. navigate to http://localhost:6006/?path=/docs/components-loading-states-bar-percent--documentation

### CODE REVIEW

File: packages/manager/src/components/BarPercent/BarPercent.tsx [hunk 1]
```diff
@@ -3,18 +3,28 @@ import { SxProps } from '@mui/system';
...
+  /** The value of the progress indicator for the determinate and buffer variants. */
+  value: number;
+  /** The value for the buffer variant. */
+  valueBuffer?: number;
 }
 
-export const BarPercent = (props: Props) => {
+/**
+ * Determinate indicator that displays how long a process will take.
+ */
```
Comment (abailly-akamai) [inline packages/manager/src/components/BarPercent/BarPercent.tsx:26]: This comment shows up as your component description in the Doc

**Review by bnussman-akamai — APPROVED** [review]
Very helpful for understanding the new patterns. Bookmarked this! Thank you

**Review by cpathipa — APPROVED** [review]
Nice work! @abailly-akamai This is really helpful and we can follow this pattern.

**Review by jaalah-akamai — APPROVED** [review]
Autodocs for the win, nice job @abailly-akamai  🚀

**Pull request merged**